torchvision/transforms/v2/_utils.py

```diff
@@ -93,3 +96,62 @@ def _check_padding_arg(padding: Union[int, Sequence[int]]) -> None:
 def _check_padding_mode_arg(padding_mode: Literal["constant", "edge", "reflect", "symmetric"]) -> None:
     if padding_mode not in ["constant", "edge", "reflect", "symmetric"]:
         raise ValueError("Padding mode should be either constant, edge, reflect or symmetric")
+
+
+def _find_labels_default_heuristic(inputs: Any) -> torch.Tensor:
+    """
+    This heuristic covers three cases:
+
+    1. The input is two-tuple whose second item is a labels tensor. This happens for already batched classification
+       inputs for Mixup and Cutmix.
+    2. The input is a two-tuple whose second item is a dictionary that contains the labels tensor under a label-like
+       (see below) key. This happens for the inputs of detection models.
+    3. The input is a dictionary that is structured as the one from 2.
+
+    What is "label-like" key? We first search for an case-insensitive match of 'labels' inside the keys of the
+    dictionary. This is the name our detection models expect. If we can't find that, we look for a case-insensitive
+    match of the term 'label' anywhere inside the key, i.e. 'FooLaBeLBar'. If we can't find that either, the dictionary
+    contains no "label-like" key.
+
+    """
+    # TODO: Document list and why
```

Comment:
Flag

Reply:
Sooo I've decided not to document it in the code because this will probably just add more confusion. But the reason we need to add support for `list` is because this is what the DataLoader actually returns (in its most default setting):

```py
for x in DataLoader(...):
    # x is  a list [img_batch, labels_batch] and we want to support
    CutMix(x)
```